[wip][generated] source: spec3.sdk.yaml@spec-cb3438d in master

mickjermsurawong-stripe opened the pull request:
r? @remi-stripe 
This is from latest spec. Does it look right to you?

remi-stripe commented on src/main/java/com/stripe/model/CreditNote.java:
```diff
@@ -0,0 +1,238 @@
+   * `order_change`, `other`, or `product_unsatisfactory`.
+   */
+  @SerializedName("reason")
+  String reason;
```
This is not correct because it hasn't been updated per the latest API review yet.

remi-stripe commented on src/main/java/com/stripe/model/Customer.java:
```diff
@@ -274,6 +274,28 @@ public Customer delete(Map<String, Object> params, RequestOptions options)
+            String.format("/v1/customers/%s/customer_balance_transactions", this.getId()));
+    return requestCollection(url, params, CustomerBalanceTransactionCollection.class, options);
+  }
+
```
Flagging that since CreditNote and BalanceTransaction are not going to be released at the same time, we will need to do 2 separate PRs.

remi-stripe commented on src/main/java/com/stripe/model/CustomerBalanceTransaction.java:
```diff
@@ -0,0 +1,117 @@
+// Generated by com.stripe.generator.entity.SdkBuilder
```
same as above, this will need to be a separate PR

remi-stripe commented on src/main/java/com/stripe/model/CustomerCollection.java:
```diff
@@ -2,4 +2,28 @@
+        String.format("%s%s", Stripe.getApiBase(), String.format("%s/%s", this.getUrl(), id));
+    return ApiResource.request(ApiResource.RequestMethod.GET, url, params, Customer.class, options);
+  }
+}
```
Can you clarify what is happening here? Looks like we put the methods on the wrong collection?

remi-stripe commented on src/main/java/com/stripe/model/PaymentIntent.java:
```diff
@@ -74,7 +74,7 @@
   String clientSecret;
 
-  /** Confirmation method of this PaymentIntent, one of `secret` or `publishable`. */
+  /** Confirmation method of this PaymentIntent, one of `manual` or `automatic`. */
```
This is kinda secret and should not be in this PR today (not a big deal though)

mickjermsurawong-stripe commented:
Closing this PR because we will release credit notes and customer balance txn separately.
Addressed the comment on `CustomerCollection` with updating pay-server client metadata on pay-server #140820
Thank you @remi-stripe!
